src/main/java/com/mixpanel/android/mpmetrics/MixpanelNotificationRouteActivity.java

```diff
@@ -136,4 +121,31 @@ public void process(MixpanelAPI api) {
         });
     }
 
+    private String getStringFromBundle(Bundle bundle, String key) {
+        String val = null;
+        CharSequence valChars = bundle.getCharSequence(key);
+        if (null == valChars) {
+            MPLog.d(LOGTAG, "$push_notification_tap logged with no \"" + key + "\" property");
+        } else {
+            val = valChars.toString();
+        }
+        return val;
+    }
+
+    private Boolean getBooleanFromBundle(Bundle bundle, String key) {
+        Boolean val = null;
+        if (!bundle.containsKey(key)) {
+            MPLog.d(LOGTAG, "$push_notification_tap logged with no \"" + key + "\" property");
+        } else {
+            val = bundle.getBoolean(key);
+        }
+        return val;
+    }
+
+    private void putIfNotNull(JSONObject json, String key, Object val) throws JSONException {
```

Comment:
can we move this to [this file](https://github.com/mixpanel/mixpanel-android/blob/master/src/main/java/com/mixpanel/android/util/JSONUtils.java)?